### PR TITLE
feat: index support for materialized views of PostgreSQL

### LINKED ISSUE #6628: Migrations skip materialized views indices
**Issue type:**

[X] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[X] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ ] `latest`
[ ] `@next`
[X] `0.2.25` (latest at the time of reporting this)

**Steps to reproduce or a small repository showing the problem:**

1. Create a materialized view with index
```ts
@ViewEntity({
  expression:
  `SELECT ...`,
  materialized: true,
})
export class UserTotal {
  @ViewColumn()
  @Index()
  userId!: string;

  ...
}
```

2. Run `typeorm migration:generate`

3. Check the migration file. It's missing `CREATE INDEX` query.

<!--
    To answer those questions you need to put "x" inside the square brackets, for example:
    [x] `mysql`
    [ ] `postgres`

    Also, please format your code properly (by taking code blocks into ```ts .... ```)
--!>

### PR DESCRIPTION
**Pull request by rodzalo**
### Description of change
Adds index support for materialized views through the `@Index()` decorator. This feature is currently available for PostgreSQL materialized views.

Fixes: [#8459](https://github.com/typeorm/typeorm/issues/8459)
Fixes #6628

### Pull-Request Checklist

- [x] Code is up-to-date with the `master` branch
- [x] `npm run format` to apply prettier formatting
- [ ] `npm run test` passes with this change
- [x] This pull request links relevant issues as `Fixes #0000`
- [x] There are new or updated unit tests validating the change
- [x] Documentation has been updated to reflect this change
- [x] The new commits follow conventions explained in [CONTRIBUTING.md](https://github.com/typeorm/typeorm/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: src/driver/postgres/PostgresQueryRunner.ts [hunk 1]
```diff
@@ -2924,6 +2944,18 @@ export class PostgresQueryRunner
...
 
+    /**
+     * Creates new view indices
+     */
+    async createViewIndices(
+        viewOrName: View | string,
+        indices: TableIndex[],
+    ): Promise<void> {
+        for (const index of indices) {
+            await this.createViewIndex(viewOrName, index)
```
Comment (dududuhyunkim) [inline src/driver/postgres/PostgresQueryRunner.ts:2955]: ?

**Comment by ghtaylor** [comment]
Would be great if we could get this sorted! Will try and take a look at the failing tests at some point

**Comment by rodzalo** [comment]
@ghtaylor Thanks for your interest in this PR! As you can see, I've resolved the issues that caused the checks to fail. If you want to help me to push this PR, you can run `npm run test` in a Docker container to check if the tests pass with this change, as my PC is not powerful enough to run Docker containers. That's the only thing that's missing in the PR checklist.

**Comment by AlexMesser** [comment]
thank you for contribution!

**Pull request merged**